demo/dlb-init.sh

```diff
@@ -0,0 +1,48 @@
+#!/bin/bash -eu
+
+enable_and_configure_vfs() {
+  devpath=$1
+  num_vfs_per_pf=$2
+  
+  sriov_numvfs_path="$devpath/sriov_numvfs"
+  if ! test -w "$sriov_numvfs_path"; then
+    echo "error: $sriov_numvfs_path is not found or not writable. Check if dlb driver module is loaded."
+    exit 1
+  fi
+  if [ "$(cat "$sriov_numvfs_path")" -ne 0 ]; then
+    echo "$devpath already configured"
+    exit 0
+  fi
+
+  #enable sriov
+  echo -n "$num_vfs_per_pf" > "$sriov_numvfs_path"
+
+  #configure vfs
+  #unbind vf
+  vf_pciid=$(basename "$(realpath "$devpath/virtfn0")")
+  dlb_pci_driver_path=/sys/bus/pci/drivers/dlb2
+  echo -n "$vf_pciid" > $dlb_pci_driver_path/unbind
+
+  #configure vf
+  vf_resources_path="$devpath/vf0_resources"
+  echo -n 2048 > "$vf_resources_path/num_atomic_inflights"
```

Comment:
I told this does not work. Please remember that we 'used to have' what you recommended and I removed.

What if the num_vfs_per_pf = 8?
The real max value of "num_sn1_slots" in simics seems to be 2.
num_sn1_slots / 8 = 0. No resource is assigned to any vf.

Reply:
the $num_vfs_per_pf is not working.. Please follow the previous comments in this thread.!

Reply:
For example, "num_sn1_slots" under total_resources shows number 8. The real maximum number seems to be 2.
If we use num_vfs_per_pf = 8, then we cannot assign any resource to num_sn1_slots.

For now, we cannot implement num_vfs_per_pf because of the limitations of simics.